From 6e9a289bd3d068f0b0a12064097e1cbfd42429c6 Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Mon, 23 Jul 2018 18:29:15 +0800
Subject: [PATCH 4240/5242] MGS-4022 [#imx-1070] fix kernel panic with opencl
 test_buffers

commit  ba964038e7ff89e4a50f9f1e30ebf5729a6b587c from
https://source.codeaurora.org/external/imx/linux-imx.git

the user memory will add the padding pages to meet hardware alignment,
need set non-contiguous flag to avoid contigous mapping in GPU MMU.

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../default/gc_hal_kernel_allocator_user_memory.c  |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_user_memory.c b/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_user_memory.c
index 223533d..4293ac9 100644
--- a/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_user_memory.c
+++ b/drivers/mxc/gpu-viv/hal/os/linux/kernel/allocator/default/gc_hal_kernel_allocator_user_memory.c
@@ -501,7 +501,8 @@ static int import_pfn_map(struct um_desc *um,
 
     Mdl->priv = userMemory;
     Mdl->numPages = userMemory->pageCount + userMemory->extraPage;
-    Mdl->contiguous = (userMemory->chunk_count == 1);
+    /* Not contiguous with extra pages */
+    Mdl->contiguous = gcvFALSE;
 
     gcmkFOOTER_NO();
     return gcvSTATUS_OK;
-- 
1.7.9.5

